src/db_options.rs

```diff
@@ -3578,6 +3582,32 @@ pub enum ChecksumType {
     XXH3 = 4, // Supported since RocksDB 6.27
 }
 
+/// Used by [`PlainTableFactoryOptions`]
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EncodingType {
```

Comment:
I guess the `EncodingType` is a too common name. I would rename it to `KeyEncodingType`.